com.reprezen.swagedit.tests/src/com/reprezen/swagedit/editor/outline/AbstractNodeTest.xtend

```diff
@@ -0,0 +1,385 @@
+	}
+
+	@Test
+	def void testGetType_Object() {
+		val text = '''
+		  info:
+		    version: 1.0.0
+		    title: hello
+		'''
+
+		val model = Model.parseYaml(text)		
+		assertNotNull(model.root)
+
+		val root = model.root
+
+		assertEquals(JsonType.OBJECT, root.type)
+		assertEquals("".ptr, root.pointer)
+
+		assertNotNull(root.get("info"))
+
+		val info = root.get("info")
+		assertEquals(JsonType.OBJECT, info.type)
+		assertEquals("info", info.property)
+		assertEquals("/info".ptr, info.pointer)
+
+		assertNotNull(info.get("version"))	
+		assertNotNull(info.get("title"))
+
+		val version = info.get("version")
+		assertEquals(JsonType.STRING, version.type)
+		assertEquals("version", version.property)
+		assertEquals("/info/version".ptr, version.pointer)
+
+		val title = info.get("title")
+		assertEquals(JsonType.STRING, title.type)
+		assertEquals("title", title.property)
+		assertEquals("/info/title".ptr, title.pointer)
+	}
+
+	@Test
+	def void testGetType_StringArray() {
+		val text = '''
+		  schemes:
+		    - http
+		    - https
+		'''
+
+		val model = Model.parseYaml(text)		
+		assertNotNull(model.root)
+
+		val root = model.root
+		
+		assertNotNull(root.get("schemes"))
+		
+		val schemes = root.get("schemes")
+		assertTrue(schemes.isArray)
+		assertEquals(JsonType.ARRAY, schemes.type)
+		assertEquals("schemes", schemes.property)
+		assertEquals("/schemes".ptr, schemes.pointer)
+		assertEquals(2, schemes.elements.size)
+		
+		val first = schemes.elements.get(0)
+		assertEquals("/schemes/0".ptr, first.pointer)
+		
+		val second = schemes.elements.get(1)
+		assertEquals("/schemes/1".ptr, second.pointer)
+	}
+
+	def ptr(String s) { JsonPointer.compile(s) }
+
+	@Test
+	def void testCreateFromArrayValues() {
+		val text = '''
+		  foo:
+		    - hello
+		    - world
+		'''
+
+		val model = Model.parseYaml(text)
+		val el = model.root.get(0)
+
+		assertEquals("foo", el.text)
+		assertEquals(2, el.elements.size)
+		
+		assertEquals("hello", el.elements.get(0).text)
+		assertEquals("world", el.elements.get(1).text)
+
+		val doc = new SwaggerDocument
+		doc.set(text)
+
+		val position = el.getPosition(doc)
+		// after foo:
+		assertEquals(4, position.offset)
+		assertEquals(0, position.length)
+
+		// should be first line
+		assertEquals(0, doc.getLineOfOffset(position.offset))
+	}
+
+	@Test
+	def void testCreateFromObjectValues() {
+		val text = '''
+		  foo: 
+		    k1: hello
+		    k2: world
+		'''
+
+		val model = Model.parseYaml(text)
+		val el = model.root.get(0)
+
+		assertEquals("foo", el.text)
+		assertEquals(2, el.elements.size)
+
+		assertEquals("k1: hello", el.elements.get(0).text)
+		assertEquals("k2: world", el.elements.get(1).text)
+
+		val doc = new SwaggerDocument
+		doc.set(text)
+
+		val position = el.getPosition(doc)
+		// after foo:
+		assertEquals(4, position.offset)
+		assertEquals(0, position.length)
+
+		// should be first line
+		assertEquals(0, doc.getLineOfOffset(position.offset))
+	}
+	
+//	@Test
```

Comment:
The `@Test` annotation is comment. Is it because the implementation of the **test** is incomplete or because of  the code? Any reason or not using `@Ignore`?